parsec/interfaces/ptg/ptg-compiler/jdf2c.c

```diff
@@ -7099,28 +7128,6 @@ jdf_generate_code_complete_hook(const jdf_t *jdf,
 
     coutput("parsec_data_t* data_t_desc = NULL; (void)data_t_desc;\n");
 
-    for( di = 0, fl = f->dataflow; fl != NULL; fl = fl->next, di++ ) {
-        if(JDF_FLOW_TYPE_CTL & fl->flow_flags) continue;
-        if(fl->flow_flags & JDF_FLOW_TYPE_WRITE) {
-            /**
-             * The data_out might be NULL if we don't forward anything.
-             */
-            coutput("  if ( NULL != this_task->data._f_%s.data_out ) {\n"
-                    "#if defined(PARSEC_DEBUG_NOISIER)\n"
-                    "     char tmp[128];\n"
-                    "#endif\n"
-                    "     this_task->data._f_%s.data_out->version++;  /* %s */\n"
```

Comment:
This is a significant change, you move the version management from `_complete`, aka. after the body execution into the `_hook`, aka. for each execution. How would that work if the body returns AGAIN or NEXT ?